File: app/src/javax/java/com/adawolfs/mvp/mvp/BaseMvpPresenterImpl.java
package com.adawolfs.mvp.mvp;

/**
 * Created by adawolfs on 1/20/17.
 */

public class BaseMvpPresenterImpl<V extends BaseMvpView> implements BaseMvpPresenter<V> {

    private V mView = null;

    @Override
    public void attachView(V view) {
        mView = view;
    }

    @Override
    public void detachView() {
        mView = null;
    }

    @Override
    public V getView() {
        return mView;
    }
}

File: app/src/kotlin/java/com/adawolfs/mvp/mvp/BaseMvpPresenterImpl.kt
package com.adawolfs.mvp.mvp

/**
 * Created by adawolfs on 1/20/17.
 */
open class BaseMvpPresenterImpl<V : BaseMvpView>(view: V) : BaseMvpPresenter<V> {

    private var mView : V? = view

    override fun attachView(view: V) {
        mView = view
    }

    override fun detachView() {
        mView = null
    }

    override fun getView(): V? {
        return mView
    }

}